### PR TITLE
Fix unexecuted migration instructions; partially fixes #85

### LINKED ISSUE #85: Review plugin migration
Plugin migrations are not managed correctly:
 - some migrations are done by executing direct queries,
 - migration is sometime executed prior to other changes,
 - migration is build via multiple fuinctions but is not executed at the end of process.

By the way, the usage of the `plugin_datainjection_needUpdateOrInstall` function is not easy to read.

### PR DESCRIPTION
**Pull request by cedric-anne**

### CODE REVIEW

**Comment by amma35** [comment]
I just tested the modifications, I get this error: 

2.5.1 multiple alter in glpi_plugin_datainjection_models - Error during query execution : ALTER TABLE `glpi_plugin_datainjection_models` CHANGE `date_mod` `date_mod` datetime NOT NULL DEFAULT NOW - Error is syntax error near'NOW' at line 1

**Comment by cedric-anne** [comment]
I should have fixed it with https://github.com/pluginsGLPI/datainjection/pull/89.

File: hook.php [hunk 1]
```diff
@@ -45,19 +45,20 @@ function plugin_datainjection_registerMethods() {
 
 
 function plugin_datainjection_install() {
-    global $DB;
+   global $DB;
+
+   include_once GLPI_ROOT."/plugins/datainjection/inc/profile.class.php";
```
Comment (trasher) [inline hook.php]: huh? why?

**Comment by cedric-anne** [comment]
@amma35 Sorry for the missed bugs. I tested twice on my installation and it works as it should. Could you valdiate on your side ?

Regards

**Comment by amma35** [comment]
@cedric-anne No problem, I'll test it as soon as I can.

**Comment by amma35** [comment]
@cedric-anne  I just tested, there's no more mistakes, it's good for me.

**Pull request merged**